Assets/Scripts/Game/StringTableCSVParser.cs

```diff
@@ -84,28 +84,19 @@ public class StringTableCSVParser
     /// <returns>KeyValuePair for each row.</returns>
     static KeyValuePair<string, string>[] ParseCSVRows(string csvText)
     {
-        // Regex pattern from https://gist.github.com/awwsmm/886ac0ce0cef517ad7092915f708175f
-        const string linePattern = "(?:,|\\n|^)(\"(?:(?:\"\")*[^\"]*)*\"|[^\",\\n]*|(?:\\n|$))";
+        // Regex pattern inspired by https://gist.github.com/awwsmm/886ac0ce0cef517ad7092915f708175f
+        // but without the exponential behavior
+        const string linePattern = "(?:\\n|^)([^\",\\n]*),((?:\"[^\"]*\")+|[^\",\\n]*)";
 
         // Split source CSV based on regex matches
-        char[] trimChars = { '\r', '\n', '\"', ',' };
-        List<KeyValuePair<string, string>> rows = new List<KeyValuePair<string, string>>();
-        string[] matches = (from Match m in Regex.Matches(csvText, linePattern, RegexOptions.ExplicitCapture) select m.Groups[0].Value).ToArray();
-        int pos = 0;
-        while (pos < matches.Length)
-        {
-            if (pos + 1 == matches.Length)
-            {
-                // Exit if no valid pair at end of csv (likely an empty line at end of source data)
-                break;
-            }
-            string key = matches[pos++].Trim(trimChars);
-            string value = matches[pos++].Trim(trimChars);
-            value = value.Replace("\"\"", "\""); // Replace escaped quotes in value with single quote marks
-            KeyValuePair<string, string> kvp = new KeyValuePair<string, string>(key, value);
-            rows.Add(kvp);
-        }
-
+        char[] trimChars = { '\r', '\n' };
```

Comment:
This could be moved outside the method so it doesn't have to allocate a new array each time